packages/compass-aggregations/src/components/aggregation-side-panel/stage-wizard-use-cases/project/project.tsx

```diff
@@ -0,0 +1,158 @@
+
+// Components
+const formGroupStyles = css({
+  display: 'flex',
+  justifyContent: 'flex-start',
+  alignItems: 'center',
+  gap: spacing[2],
+});
+
+const labelStyles = css({
+  minWidth: '140px',
+  textAlign: 'right',
+});
+
+const containerStyles = css({
+  display: 'flex',
+  flexDirection: 'column',
+  gap: spacing[2],
+});
+
+export const ProjectForm = ({
+  fields,
+  variant = 'include',
+  onChange,
+}: ProjectProps) => {
+  const [projectFields, setProjectFields] = useState<ProjectFormState>([null]);
+  const comboboxStyles = useMemo(() => {
+    const placeholderLength = PLACEHOLDER_TEXT.length;
+    return {
+      width: `calc(${String(
+        Math.max(placeholderLength, ...fields.map((label) => label.length)) + 1 // +1 is to ensure some extra buffer on right
+      )}ch)`,
+    };
+  }, [fields]);
+
+  useEffect(() => {
+    const stageValue = mapProjectFormStateToStageValue(variant, projectFields);
+
+    onChange(
+      JSON.stringify(stageValue),
+      Object.keys(stageValue).length === 0
+        ? new Error('No field selected')
+        : null
+    );
+  }, [projectFields, onChange, variant]);
+
+  const variantText = variant === 'include' ? 'Include' : 'Exclude';
+
+  const onSelectField = (index: number, value: string | null) => {
+    if (!value) return;
+    const nextProjectFields = [...projectFields];
+    nextProjectFields[index] = value;
+    setProjectFields(nextProjectFields);
+  };
+
+  const addItem = (at: number) => {
+    const nextProjectFields = [...projectFields];
+    nextProjectFields.splice(at + 1, 0, null);
+    setProjectFields(nextProjectFields);
+  };
+
+  const removeItem = (at: number) => {
+    const nextProjectFields = [...projectFields];
+    nextProjectFields.splice(at, 1);
+    setProjectFields(nextProjectFields);
+  };
+
+  return (
+    <div className={containerStyles}>
+      {projectFields.map((fieldValue, index) => (
+        <div
+          className={formGroupStyles}
+          key={`project-${variant}-form-${index}`}
+          data-testid={`project-${variant}-form-${index}`}
+        >
+          <Body className={labelStyles}>
+            {index === 0 ? variantText : 'and'}
+          </Body>
+          <div data-testid={`project-${variant}-form-${index}-field`}>
+            <ComboboxWithCustomOption
+              placeholder={PLACEHOLDER_TEXT}
+              style={comboboxStyles}
+              aria-label={PLACEHOLDER_TEXT}
+              size="default"
+              clearable={false}
+              value={fieldValue}
+              onChange={(value: string | null) => onSelectField(index, value)}
+              options={fields}
+              optionLabel="Field:"
+              // Used for testing to access the popover for a stage
+              popoverClassName={`project-${variant}-form-${index}-field-combobox`}
+            />
+          </div>
+          <IconButton aria-label="Add" onClick={() => addItem(index)}>
+            <Icon color={palette.black} glyph="Plus" />
```

Comment:
do we need to specify the color here? if so, we have to handle it in the dark mode as well.

Reply:
Ahh, good point! I specified it because the default color was grayish and in the design it was black. I will handle the dark mode as well here.

Reply:
Just added it.